### ISSUE TITLE
using mod storage instead of txt

### ISSUE THREAD

**Issue by MoNTE48** [FaceDeer/settlements#3]
So this is implemented in my fork of settlements.
Much better than filling the world folder with countless txt files from each mod.

```
local mod_storage = minetest.get_mod_storage()
-------------------------------------------------------------------------------
-- save list of generated villages
-------------------------------------------------------------------------------
function villages.save()
	mod_storage:set_string("villages_in_world", minetest.serialize(villages_in_world))
end
-------------------------------------------------------------------------------
-- load list of generated villages
-------------------------------------------------------------------------------
function villages.load()
	return minetest.deserialize(mod_storage:get_string("villages_in_world")) or {}
end
```

**Comment by FaceDeer** [comment]
I'm a little wary of this. A major obstacle I ran into with Digtron is that I ran into a limit on how large a metadata string could get when serializing a large Digtron layout table into a metadata value, have you stress-tested this with an extremely large set of villages? Digtron was using item metadata, not mod storage, so maybe that's not an issue here, I don't know.

Also, I'm using an AreaStore object for tracking settlement locations and it's got its own to_string() and from_string(str) serialization methods so I would have thought there'd be something funky about using regular serialization with them. I guess minetest.serialize works after all, though? Good to know.

Edit: Oh, wait, your example code is "villages" rather than "settlements" -  is that a fork from before I did my big revamp and switched over to AreaStore? The settlement list was just a simple list of vectors then, so maybe minetest.serialize doesn't work with AreaStore after all. Would need to test.

Anyway, I'm not declining this suggestion, just not going to rush into it right now - I'm worn out on this mod for now and don't want to rip out a working system to dive into the unknown again at the moment. :) All the persistence stuff should be isolated inside persistence.lua so it shouldn't be messy to fix later.

**Comment by FaceDeer** [comment]
I think I'll have to close this one as cannot-fix, alas. I did a bit of experimenting and it seems that the string that AreaStore:to_string() is a binary string that mod storage isn't able to handle properly. This matches the documentation:

"Returns area store serialized as a (binary) string." (from the AreaStore to_string() doc) and "WARNING: This storage backend is incaptable to save raw binary data due to restrictions of JSON." (from the mod metadata StorageRef documentation)

I've moved all this stuff over into a separate mod, too: https://github.com/FaceDeer/named_waypoints
So if a better method of persisting this data comes up there's a central location to fix it in.
